tree/tree/inc/ROOT/RFriendInfo.hxx

```diff
@@ -58,13 +63,30 @@ struct RFriendInfo {
     */
    std::vector<std::vector<std::int64_t>> fNEntriesPerTreePerFriend;
 
-   void AddFriend(const std::string &treeName, const std::string &fileNameGlob, const std::string &alias = "");
+   /**
+      Information on the friend's TTreeIndexes.
+   */
+   std::vector<std::unique_ptr<TVirtualIndex>> fTreeIndexInfos;
+
+   RFriendInfo() = default;
+   RFriendInfo(const RFriendInfo &);
+   RFriendInfo &operator=(const RFriendInfo &);
+   RFriendInfo(RFriendInfo &&) = default;
+   RFriendInfo &operator=(RFriendInfo &&) = default;
+   RFriendInfo(std::vector<std::pair<std::string, std::string>> friendNames,
+               std::vector<std::vector<std::string>> friendFileNames,
+               std::vector<std::vector<std::string>> friendChainSubNames,
+               std::vector<std::vector<std::int64_t>> nEntriesPerTreePerFriend,
+               std::vector<std::unique_ptr<TVirtualIndex>> treeIndexInfos);
```

Comment:
I guess this constructor is needed to allow for the other constructors/assignment operators right? In principle, if copy semantics are removed, all the constructors could be removed too IIUC

Reply:
> this constructor is needed to allow for the other constructors/assignment operators right?

No, this constructor is needed by `GetFriendInfo`. The other constructors or assignment operators can live without this.

Reply:
It's just that previously there were no constructors so I was wondering why now we need one.

Reply:
`RFriendInfo` now has a data member of type `vector<unique_ptr<_>>` which implicitly disables copy-construction and copy-assignment, so I implement them explicitly in order to not break backwards-compatibility. And implementing other constructors explicitly disables aggregate initialization (`RFriendInfo{x, y, z}`), so we need a dedicated constructor for that purpose.